File: Strings/1_10/9Solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max=0;
        Set<Character> set=new HashSet<>();
        int start=0,end=0;
        while(end<s.length()){
            if(!set.contains(s.charAt(end))){
                set.add(s.charAt(end++));
                max=Math.max(max,set.size());
            }else{
                set.remove(s.charAt(start));
                start++;
            }
        }
        return max;
    }
}

File: Strings/1_10/9Solution.py
def lengthOfLongestSubstring(s):
        dic = {}
        ans = 0
        tmp = 0
        start = 0
        for i in range(len(s)):
            if dic.get(s[i], -1)==-1:
                dic[s[i]] = i
                tmp+=1
                ans = max(ans,tmp)
            else:
                if dic[s[i]] >= start:
                    tmp = tmp - (dic[s[i]] - start)
                    start = dic[s[i]]+1
                    dic[s[i]] = i
                else:
                    dic[s[i]] = i
                    tmp +=1
                    ans = max(ans, tmp)
        return ans
s=input()
print(lengthOfLongestSubstring(s))